### PR TITLE
Fix application context

### PR DESCRIPTION
**Pull request by Fei-Chen-**

### CODE REVIEW

**Comment by smokienko** [comment]
If we are changing everithing to Activity instead of application, than we should change strong refferences to weak refferences, othervise we risk to leak activities, especially in current architecture, because OktaAppAuth is a singletone.

**Comment by FeiChen-okta** [comment]
### Problem Analysis (Technical)
* Endsession doesn't do anything because the context used to start the browser tab is of type Application instead of Activity. This will case a Calling startActivity() from outside of an Activity context exception. This works for login because OktaManagementActivity is using its own Activity context  passed to AppAuth.
  Related issue: #86 

### Solution (Technical)
* This is a bug in OktaAppAuth and AppAuth SDK. Short term solution is to remove context.getApplicationContext in OktaAppAuth constructor. Long term solution is to make sure the context is always of type Activity when starting custom tabs. This should be done in the next revision.

### Affected Components
Endsession flow and any other flow that requires custom tabs.

### Tests
On emulator with API level 28.
On emulator with API level 22.
On emulator with API level 16.

File: library/src/main/java/com/okta/appauth/android/OktaAppAuth.java [hunk 1]
```diff
@@ -100,7 +100,7 @@
     public static OktaAppAuth getInstance(@NonNull Context context) {
```
Comment (smokienko) [inline library/src/main/java/com/okta/appauth/android/OktaAppAuth.java]: As long as context is a WeakReference now I think we can add context reinitialization if it is null. Something like:
 OktaAppAuth oktaAppAuth = INSTANCE_REF.get().get();
```
        if (oktaAppAuth == null) {
            oktaAppAuth = new OktaAppAuth(context);
            INSTANCE_REF.set(new WeakReference<>(oktaAppAuth));
        } else if (oktaAppAuth.mContext.get() == null) {
            oktaAppAuth.mContext = new WeakReference<>(context);
        }

        return oktaAppAuth;

```

File: library/src/main/java/com/okta/appauth/android/OktaAppAuth.java [hunk 2]
```diff
@@ -741,7 +744,7 @@ private void recreateAuthorizationService(Context context) {
      */
     AuthorizationService createAuthorizationServiceIfNeeded() {
         if (mAuthService.get() == null) {
-            recreateAuthorizationService(mContext);
+            recreateAuthorizationService(mContext.get());
```
Comment (bdemers) [inline library/src/main/java/com/okta/appauth/android/OktaAppAuth.java]: does `recreateAuthorizationService` work if the context is null here?

Comment (FeiChen-okta) [reply library/src/main/java/com/okta/appauth/android/OktaAppAuth.java]: It won't work if context is null. This only happens when the Activity is closed. This should never happen but I'll add a null check and throw a exception to init OktaAppAuth again.

**Pull request merged**